src/nmodl/nocpout.cpp

```diff
@@ -573,15 +589,18 @@ extern Memb_func* memb_func;\n\
                 thread_data_index);
         Lappendstr(defs_list, buf);
         Sprintf(buf,
-                " if (_thread1data_inuse) {_thread[_gth] = (double*)ecalloc(%d, "
-                "sizeof(double));\n }else{\n _thread[_gth] = &_thread1data[0]; "
-                "_thread1data_inuse = 1;\n }\n",
+                "if (_thread1data_inuse) {\n"
+                "  _thread[_gth] = {neuron::container::do_not_search, new double[%d]{}};\n"
```

Comment:
```cpp
     * Unlike the constructor taking T*, this does *not* attempt to promote raw pointers to modern
     * data_handles.
     */
    data_handle(do_not_search_t, T* raw_ptr)
```
just to clarify: so these stores just raw, allocated ptr inside data handle as it is? and then would be a common pattern for all raw ptrs handling?

Reply:
Yes, it stores a raw pointer without any special handling.

Ideally we would distinguish between "real" `POINTER` variables to other parts of the model, which need `data_handle<double>` from "raw" `POINTER` variables (the ones we use via `_p_XXX`) as the former is overkill for the latter

`sizeof(data_handle<double>)` is typically `3*sizeof(double*)` and `sizeof(Datum (= generic_data_handle))` is typically `4*sizeof(double*)`; plus the logic is more complicated.